### PR TITLE
feat: apiVersion from config

### PR DESCRIPTION
**Pull request by srchase**
Get apiVersion from resolved config for Glacier instead of using hard-coded value.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/middleware-sdk-glacier/src/configurations.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+interface PreviouslyResolved {
+  apiVersion: string;
+}
+
+export interface ResolvedAddGlacierApiVersionMiddlewareConfig {
+  apiVersion: string;
+}
+
+export function resolveSsecMiddlewareConfig<T>(
```
Comment (AllanFly120) [inline packages/middleware-sdk-glacier/src/configurations.ts]: If this plugin is called `GlacierCustomizationsPlugin` these interfaces should be `GlacierCustomizationsResolvedConfig` and `GlacierCustomizationInputConfig`

Comment (srchase) [reply packages/middleware-sdk-glacier/src/configurations.ts]: fixed in 1964189290c898e22dce6a0bd25d206bf0e20e7f

File: packages/middleware-sdk-glacier/src/add-glacier-api-version.spec.ts [hunk 2]
```diff
@@ -9,10 +9,9 @@ describe("addGlacierApiVersion", () => {
   });
 
   it("sets the x-amz-glacier-version header", async () => {
-    const handler = addGlacierApiVersionMiddleware()(
-      mockNextHandler,
-      {} as any
-    );
+    const handler = addGlacierApiVersionMiddleware({
```
Comment (AllanFly120) [inline packages/middleware-sdk-glacier/src/add-glacier-api-version.spec.ts]: If the function already lives in a `glacier` package, you don't need to mention `glacier` in your exported variable name again

**Comment by srchase** [comment]
Moved these changes into https://github.com/aws/aws-sdk-js-v3/pull/548

**Pull request closed without merging**